apps/rush/src/actions/BaseAction.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import {
+  CommandLineAction,
+  ICommandLineActionOptions
+} from '@microsoft/ts-command-line';
+
+import {
+  RushConfiguration
+} from '@microsoft/rush-lib';
+
+import EventHooksManager from '../utilities/EventHooksManager';
+
+export abstract class BaseAction extends CommandLineAction {
+  private _rushConfiguration: RushConfiguration;
+  private _eventHooksManager: EventHooksManager;
+
+  constructor(options: ICommandLineActionOptions) {
+    super(options);
+  }
+
+  protected onExecute(): void {
+    this._ensureEnvironment();
+    this.run();
+  }
+
+  protected abstract run(): void;
+
+  protected get rushConfiguration(): RushConfiguration {
+    if (!this._rushConfiguration) {
+      this._rushConfiguration = RushConfiguration.loadFromDefaultLocation();
```

Comment:
I feel it is actually natural if it throws whenever someone happens to look at rush configuration and there is mistake in rush.json. 

---
In reply to: [123096996](https://github.com/Microsoft/web-build-tools/pull/243#discussion_r123096996) [](ancestors = 123096996)